Administrator Guide/Consul.md

```diff
@@ -0,0 +1,607 @@
+
+Next, we must define gluster service consul on the servers.
+
+## Consul agent on Linux on gluster servers
+
+Install consul agent as described in previous section.
+
+You can define consul services as ``gluster`` to run health checks, to do that we must add consul to sudoers or allow it executing certain sudo commands without password:
+
+``/etc/sudoers.d/99-consul.conf``:
+
+```text
+
+    consul ALL=(ALL) NOPASSWD: /sbin/gluster pool list
+
+```
+
+First, lets define service in consul, it will be very basic, without volume names.
+Service name ``gluster``, with default port 24007, and we will tag it as ``gluster`` and ``server``.
+
+Our service will have [service health checks](https://www.consul.io/docs/agent/checks.html) every 10s:
+
+* check if the gluster service is responding to TCP on 24007 port
+* check if the gluster server is connected to other peers in the pool (to avoid registering as healthy service which is actaully not serving anything)
+
+Below is an example of ``/etc/consul/service_gluster.json``:
+
+```json
+
+    {
+      "service": {
+        "address": "",
+        "checks": [
+          {
+            "interval": "10s",
+            "tcp": "localhost:24007",
+            "timeout": "5s"
+          },
+          {
+            "interval": "10s",
+            "script": "/bin/bash -c \"sudo -n /sbin/gluster pool list |grep -v UUID|grep -v localhost|grep Connected\"",
+            "timeout": "5s"
+          }
+        ],
+        "enableTagOverride": false,
+        "id": "gluster",
+        "name": "gluster",
+        "port": 24007,
+        "tags": [
+          "gluster",
+          "server"
+        ]
+      }
+    }
+
+```
+
+Restart consul service and you should see gluster servers in consul web ui.
+After a while service should be in healthy stage and be available under nslookup:
+
+```bash
+
+    [centos@gluster-poc-02]# nslookup gluster.service.consul
+    Server:     127.0.0.1
+    Address:    127.0.0.1#53
+
+    Name:   gluster.service.consul
+    Address: 172.30.64.144
+    Name:   gluster.service.consul
+    Address: 172.30.65.61
+
+```
+
+Notice that gluster server can be also gluster client, for example if we want to mount gluster volume on the servers.
+
+## Mounting gluster volume under Linux
+
+As a moutpoint we would usually select one of the gluster servers, and another as backup server, like this:
+
+```bash
+
+    mount -t glusterfs -o backupvolfile-server=gluster-poc-02 gluster-poc-01:/g0 /mnt/gluster/g0
+
+```
+
+This is a bit inconvenient, for example we have an image with hardcoded hostnames, and old servers are gone due to maintenance.
+We would have to recreate image, or reconfigure existing nodes if they unmount gluster storage.
+
+To mitigate that issue we can now use consul for fetching the server pool:
+
+```bash
+
+    mount -t glusterfs gluster.service.consul:/g0 /mnt/gluster/g0
```

Comment:
Ah, I see. I'm new to consul and this is where I realise what this document is all about. Can you briefly add what consul integration solves for the users at the beginning of this doc ?

Thanks for the detailed PR 👍